task/vpd/src/main.rs

```diff
@@ -18,6 +18,25 @@ struct ServerImpl;
 
 task_slot!(I2C, i2c_driver);
 
+fn eeprom_is_locked(
+    dev: &drv_i2c_devices::at24csw080::At24Csw080,
+) -> Result<bool, RequestError<VpdError>> {
+    use drv_i2c_devices::at24csw080::WriteProtectBlock;
+
+    match dev.read_eeprom_write_protect() {
+        Err(drv_i2c_devices::at24csw080::Error::I2cError(code)) => {
+            let err: VpdError = code.into();
+            Err(err.into())
+        }
+        Err(_) => Err(VpdError::DeviceError.into()),
+        Ok(wp) if wp.locked => match wp.block {
+            Some(WriteProtectBlock::AllMemory) => Ok(true),
+            _ => Err(VpdError::PartiallyLocked.into()),
+        },
+        _ => Ok(false),
```

Comment:
Nit:
```suggestion
        Ok(_) => Ok(false),
```
this makes it more obvious that we caught all errors, without having to look up